doc/man1/flux-archive.rst

```diff
@@ -0,0 +1,314 @@
+
+   - It should have enough space to hold the extracted files.
+
+   - It should not be a fragile network file system such that parallel
+     writes could cause a distributed denial of service.
+
+   - It should not already be shared among the nodes of your job.
+
+.. option:: -v, --verbose=[LEVEL]
+
+   List files on standard error as the archive is extracted.
+
+.. option:: --overwrite
+
+   Overwrite existing files when extracting.  :program:`flux archive extract`
+   normally refuses to do this and treats it as a fatal error.
+
+.. option:: --waitcreate[=FSD]
+
+   Wait for the archive key to appear in the KVS if it doesn't exist.
+
+   If *FSD* is specified, it is interpreted as a timeout value in RFC 23
+   Flux Standard Duration format.
+
+   .. note::
+
+      The KVS employs an "eventually consistent" cache update model, which
+      means one has to be careful when writing a key on one broker rank and
+      reading it on other broker ranks.  Without some form of synchronization,
+      there is a short period of time where the KVS cache on the other ranks
+      may not yet have the new data.
+
+      This is not an issue for example 2 below, where a batch script creates
+      an archive, then submits jobs that read the archive because job
+      submission and execution already include KVS synchronization.
+      In other situations such as example 1, it is advisable to use
+      :option:`--waitcreate` or to explicitly synchronize between writing
+      the archive and reading it, e.g.
+
+      .. code-block:: console
+
+        flux exec -r all flux kvs wait $(flux kvs version)
+
+.. option:: --no-force-primary
+
+   Extract from the archive in the default KVS namespace, honoring
+   :envvar:`FLUX_KVS_NAMESPACE`, if set.  By default, the primary KVS
+   namespace is used.
+
+
+EXAMPLE
+=======
+
+Example 1:  a batch script that archives data from ``/project/dataset1``, then
+replicates it in a temporary directory on each node of the batch allocation
+where it can be used by multiple jobs.
+
+.. code-block:: console
+
+  #!/bin/bash
+
+  flux archive create -C /project dataset1
+  flux exec -r all mkdir -p /tmp/project
+  flux exec -r all flux archive extract --waitcreate -C /tmp/project
+
+  # app1 and app2 have access to local copy of dataset1
+  flux run -N1024 app1 --input=/tmp/project/dataset1
+  flux run -N1024 app2 --input=/tmp/project/dataset1
+
+  # clean up
+  flux exec -r all rm -rf /tmp/project
+  flux archive remove
+
+Example 2: a batch script that archives a large executable and a data set,
+then uses the ``stage-in`` shell plugin to copy them to
+:envvar:`FLUX_JOB_TMPDIR` which is automatically cleaned up after each job.
+
+.. code-block:: console
+
+  #!/bin/bash
+
+  flux archive create --name=dataset1 -C /project dataset1
+  flux archive create --name=app --mmap -C /home/fred app
+
+  flux run -N1024 -o stage-in.keys=app,dataset1 \
+    sh -c "\$FLUX_JOB_TMPDIR/app --input=\$FLUX_JOB_TMPDIR/dataset1"
+
```

Comment:
I noticed this while reviewing the `--waitcreate` note and thought I'd point it out before a full review. Since the job shell supports `{{tmpdir}}` replacement in command args, we may as well feature that here and simplify the example:

```suggestion
  flux run -N1024 -o stage-in.keys=app,dataset1 {{tmpdir}}/app --input={{tmpdir}}/dataset1
```

Reply:
Great idea!  I had forgotten about that actually.